### ISSUE TITLE
mailwatch_sendmail_relay.php error: Fatal error: Class 'MtaLogProcessor' not found

### ISSUE THREAD

**Issue by jakes670** [mailwatch/MailWatch#912]
### Issue summary
Relay Information from the send maillogs is not in database
When running  /usr/local/bin/mailwatch_sendmail_relay.php

get the following error

Fatal error: Class 'MtaLogProcessor' not found in /usr/local/bin/mailwatch_sendmail_relay.php on line 58

### Steps to reproduce
1. copy from tools/Sendmail_relay/mailwatch_sendmail_relay.php  to /usr/local/bin/mailwatch_sendmail_relay.php
2. chmod +x /usr/local/bin/mailwatch_sendmail_relay.php
3. now run /usr/local/bin/mailwatch_sendmail_relay.php

### Expected result
/usr/local/bin/mailwatch_sendmail_relay.php
must update database with relay information 

### Actual result
get error 
Fatal error: Class 'MtaLogProcessor' not found in /usr/local/bin/mailwatch_sendmail_relay.php on line 54

### Version and method
MailWatch Version: 1.2.4
MailScanner Version: 4.85.2
ClamAV Version: 0.99.2
SpamAssassin Version: 3.3.1
PHP Version: 5.6.31
MySQL Version: 5.6.37

### Server configuration
 - Operation System: [ CentOS ]
 - PHP version: [  5.6  ]
 - Database: [ MySQL ]
 - Mail Server: [  Sendmail ]
 - Used web server: [ Apache]

### Client configuration 
 - Operation System: [ Windows 10 ]
 - Browser: [Firefox v54 ]

**Comment by Skywalker-11** [comment]
If you change line 44 in mailwatch_sendmail_relay.php from
`$pathToFunctions = '/opt/mailwatch/mailscanner/mtalogprocessor.inc.php';` 
to 
`$pathToFunctions = MAILWATCH_HOME . '/mtalogprocessor.inc.php';` 
it should work.

Fix in #913

**Comment by jakes670** [comment]
Hi Michael

Thanks for the quick reply,  I have updated the line 44

require $pathToFunctions;

// Edit if you changed webapp directory from default
// $pathToFunctions = '/opt/mailwatch/mailscanner/mtalogprocessor.inc.php';
$pathToFunctions = MAILWATCH_HOME . '/mtalogprocessor.inc.php';

I also tried changing it to /var/www/html/mailscanner/mtalogprocessor.inc.php

I however still get the error
Fatal error: Class 'MtaLogProcessor' not found in /usr/local/bin/mailwatch_sendmail_relay.php on line 55

The file does exist

[root@smtp bin]# ls -l /var/www/html/mailscanner/mtalogprocessor.inc.php
-rw-r--r-- 1 root root 7643 Aug 17 11:59 /var/www/html/mailscanner/mtalogprocessor.inc.php



Regards
Jaco

**Comment by endelwar** [comment]
@jakes670 
add a `require $pathToFunctions;` after `$pathToFunctions = MAILWATCH_HOME . '/mtalogprocessor.inc.php';` and report if it is working

**Comment by Skywalker-11** [comment]
Can you try the version in #913 and run the file again?: 
[https://raw.githubusercontent.com/Skywalker-11/1.2.0/a2d7c05bb72b94b7deef3b16a87a1718ffda258a/tools/Sendmail_relay/mailwatch_sendmail_relay.php](https://raw.githubusercontent.com/Skywalker-11/1.2.0/a2d7c05bb72b94b7deef3b16a87a1718ffda258a/tools/Sendmail_relay/mailwatch_sendmail_relay.php)

**Comment by jakes670** [comment]
Still the same error

See below.

root@smtp ~]# cd /usr/local/bin/

[root@smtp bin]# mv mailwatch_sendmail_relay.php mailwatch_sendmail_relay.php.old

[root@smtp bin]# wget https://raw.githubusercontent.com/Skywalker-11/1.2.0/a2d7c05bb72b94b7deef3b16a87a1718ffda258a/tools/Sendmail_relay/mailwatch_sendmail_relay.php
--2017-08-18 07:37:25--  https://raw.githubusercontent.com/Skywalker-11/1.2.0/a2d7c05bb72b94b7deef3b16a87a1718ffda258a/tools/Sendmail_relay/mailwatch_sendmail_relay.php
Resolving raw.githubusercontent.com... 151.101.172.133
Connecting to raw.githubusercontent.com|151.101.172.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4507 (4.4K) [text/plain]
Saving to: “mailwatch_sendmail_relay.php”

100%[===================================================================================================================================================================================================>] 4,507       --.-K/s   in 0s

2017-08-18 07:37:26 (158 MB/s) - “mailwatch_sendmail_relay.php” saved [4507/4507]

[root@smtp bin]# chmod +x mailwatch_sendmail_relay.php

[root@smtp bin]# ./mailwatch_sendmail_relay.php

Fatal error: Class MtaLogProcessor contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (MtaLogProcessor::extractKeyValuePairs) in /var/www/html/mailscanner/mtalogprocessor.inc.php on line 232

**Comment by Skywalker-11** [comment]
Can't reproduce this with PHP5.6.29 on debian and the updated script.

**Comment by jakes670** [comment]
Seems this is an issue on Centos, I did the same test on another server running with EFA appliance, results the same below

[root@smtp3 Sendmail_relay]# ./mailwatch_sendmail_relay.php

Fatal error: Class 'MtaLogProcessor' not found in /usr/local/bin/mailwatch/tools/Sendmail_relay/mailwatch_sendmail_relay.php on line 58

Config on this server:
// Edit if you changed webapp directory from default
$pathToFunctions = '/var/www/html/mailscanner/mtalogprocessor.inc.php';

File exist in location:
[root@smtp3 Sendmail_relay]# ls -l /var/www/html/mailscanner/mtalogprocessor.inc.php
-rw-r--r-- 1 root root 7820 Apr 29 09:12 /var/www/html/mailscanner/mtalogprocessor.inc.php

Details about this second server:

[root@smtp3 Sendmail_relay]# cat /etc/issue

------------------------------
--- Welcome to EFA-3.0.2.2 ---
------------------------------
  http://www.efa-project.org
------------------------------

[root@smtp3 Sendmail_relay]# uname -a
Linux smtp3 2.6.32-696.3.2.el6.x86_64 #1 SMP Tue Jun 20 01:26:55 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

[root@smtp3 Sendmail_relay]# rpm -qa |grep php
php-5.3.3-49.el6.x86_64
php-common-5.3.3-49.el6.x86_64
php-mbstring-5.3.3-49.el6.x86_64
php-cli-5.3.3-49.el6.x86_64
php-mysql-5.3.3-49.el6.x86_64
php-pdo-5.3.3-49.el6.x86_64
php-gd-5.3.3-49.el6.x86_64

[root@smtp3 Sendmail_relay]# rpm -qa |grep sql
sqlite-3.6.20-1.el6_7.2.x86_64
php-mysql-5.3.3-49.el6.x86_64
postgresql-libs-8.4.20-7.el6.x86_64


Many thanks
Jaco

**Comment by Skywalker-11** [comment]
The error message on the second server indicates that the file is not included. Is there a `require $pathToFunctions` or `require_once $pathToFunctions` placed after the `$pathToFunctions = '/var/www/html/mailscanner/mtalogprocessor.inc.php';`

This problem with the mtalogprocessor.inc.php not beeing included was introduced with v1.2.0-rc5

**Comment by shawniverson** [comment]
I'm confused....please let me know what I am missing here....

In the latest deveop...

Line 35:  
```
$pathToFunctions = '/var/www/html/mailscanner/functions.php';
```

Line 44:
```
$pathToFunctions = '/opt/mailwatch/mailscanner/mtalogprocessor.inc.php';
```

1) Why is the same variable being reused?
2) Line 44 is never followed by a require or require_once.

**Comment by shawniverson** [comment]
I see #913 addresses that...

**Comment by stefaweb** [comment]
Hello @shawniverson!

`/var/www/html/mailscanner` is for the current MailWatch version.

`/opt/mailwatch/` are scories from a previous @endelwar PR (`opt` will be used with MailWatch 2.0).

All ref to `/opt/mailwatch/` should be replaced with /var/www/html/mailscanner in current dev code.

**Comment by Skywalker-11** [comment]
@stefaweb as addition to your statement:
Usually we don't need to use the full path when including files but just include the function.php once that way. After that we can use the MAILWATCH_HOME constant to include everything else. That we can keep the static imports as low as possible.

Found another occurrence of /opt/mailwatch in postfix_relay. Fixed in #913
